src/sentry/integrations/github/integration.py

```diff
@@ -929,7 +929,14 @@ def dispatch(self, request: HttpRequest, pipeline: Pipeline) -> HttpResponseBase
                 if chosen_installation_id == "-1":
                     return pipeline.next_step()
 
-                if not has_scm_multi_org:
+                # Validate the same org is installing and that they have the multi org feature
+                installing_organization_slug = pipeline.fetch_state("installing_organization_slug")
+                is_same_installing_org = (
+                    (installing_organization_slug is not None)
+                    and installing_organization_slug
+                    == self.active_user_organization.organization.slug
+                )
+                if not has_scm_multi_org or not is_same_installing_org:
```

Comment:
Good catch! But should we handle this with a separate error more generally, like break the pipeline if the user is seeing resources from a different organization after re-fetching? I'll leave it to you but I think that switching organizations mid installation is totally fine to break the installation flow.

Reply:
ooo that makes sense uh, I just would need to figure out how to do that in the frontend :hides: